nibabel/tests/test_surfaceimages.py

```diff
@@ -0,0 +1,221 @@
+        with file_map['image'].get_prepare_fileobj('wb') as fobj:
+            np.save(fobj, self.arr)
+
+    def get_data_dtype(self):
+        return self.dataobj.dtype
+
+    def set_data_dtype(self, dtype):
+        self.dataobj = self.dataobj.astype(dtype)
+
+
+class FSGeometryProxy:
+    def __init__(self, pathlike):
+        self._file_like = str(Path(pathlike))
+        self._offset = None
+        self._vnum = None
+        self._fnum = None
+
+    def _peek(self):
+        from nibabel.freesurfer.io import _fread3
+        with open(self._file_like, "rb") as fobj:
+            magic = _fread3(fobj)
+            if magic != 16777214:
+                raise NotImplementedError("Triangle files only!")
+            fobj.readline()
+            fobj.readline()
+            self._vnum = np.fromfile(fobj, ">i4", 1)[0]
+            self._fnum = np.fromfile(fobj, ">i4", 1)[0]
+            self._offset = fobj.tell()
+
+    @property
+    def vnum(self):
+        if self._vnum is None:
+            self._peek()
+        return self._vnum
+
+    @property
+    def fnum(self):
+        if self._fnum is None:
+            self._peek()
+        return self._fnum
+
+    @property
+    def offset(self):
+        if self._offset is None:
+            self._peek()
+        return self._offset
+
+    @property
+    def coords(self):
+        ap = ArrayProxy(self._file_like, ((self.vnum, 3), ">f4", self.offset))
+        ap.order = 'C'
+        return ap
+
+    @property
+    def triangles(self):
+        offset = self.offset + 12 * self.vnum
+        ap = ArrayProxy(self._file_like, ((self.fnum, 3), ">i4", offset))
+        ap.order = 'C'
+        return ap
+
+
+class FreeSurferHemisphere(SurfaceGeometry):
+    @classmethod
+    def from_filename(klass, pathlike):
+        path = Path(pathlike)
+        hemi, default = path.name.split(".")
+        mesh_names = ('orig', 'white', 'smoothwm',
+                      'pial', 'inflated', 'sphere',
+                      'midthickness', 'graymid')  # Often created
+        if default not in mesh_names:
+            mesh_names.append(default)
+        meshes = {}
+        for mesh in mesh_names:
+            fpath = path.parent / f"{hemi}.{mesh}"
+            if fpath.exists():
+                meshes[mesh] = FSGeometryProxy(fpath)
+        hemi = klass(meshes)
+        hemi._default = default
+        return hemi
+
+    def get_coords(self, name=None):
+        if name is None:
+            name = self._default
+        return self._meshes[name].coords
+
+    def get_triangles(self, name=None):
+        if name is None:
+            name = self._default
+        return self._meshes[name].triangles
+
+    @property
+    def n_coords(self):
+        return self.meshes[self._default].vnum
+
+    @property
+    def n_triangles(self):
+        return self.meshes[self._default].fnum
+
+
+class FreeSurferSubject(GeometryCollection):
+    @classmethod
+    def from_subject(klass, subject_id, subjects_dir=None):
```

Comment:
So far:
```
>>> from nibabel.tests.test_surfaceimages import FreeSurferSubject
>>> subj = FreeSurferSubject.from_subject('fsaverage', '/mnt/bakraid/data/structurals')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/larsoner/python/nibabel/nibabel/tests/test_surfaceimages.py", line 210, in from_subject
    return klass.from_directory(Path(subjects_dir) / subject_id)
AttributeError: type object 'FreeSurferSubject' has no attribute 'from_directory'
```
So some unit tests are clearly needed :)

And if I change the call to use `from_spec`, then there is no `self` referred to in that class method